packages/drivers/odsp-driver/src/odspDriverUrlResolver.ts

```diff
@@ -124,9 +132,13 @@ export async function resolveRequest(request: IRequest): Promise<IOdspResolvedUr
             itemId: "",
             fileName,
             summarizer: false,
+            containerPackage: {
+                packageName,
+            },
         };
     }
     const { siteUrl, driveId, itemId, path } = decodeOdspUrl(request.url);
```

Comment:
Where is the magic that encodes and decodes things in the nav parameter for ODSP urls? Would expect we need to modify that to put the code hint in the URL and to read it off the URL

Reply:
I think this branch will never have ContainerPackageHeader.containerPackage, instead containerPackageName  should come from decodeOdspUrl?
As for nav parameter, parsing it happens before we get here, right?

Reply:
According to this logic, we are supplying the  package name as part of request headers and not as part of url.

Reply:
I guess I'm still confused, I don't see anything in the PR around nav parameter encoding/decoding. Is this something that happens in the FFX layer today?

Reply:
Is there any concern about using headers instead of adding the information to the URL itself? we can totally change it to be in the URL, i'm not against it, it only seemed cleaner to me.

Reply:
I'm expecting this to end up in the real URL, which makes me favor having this be in the URL rather than a header. But then again, I'm not super informed on the tradeoffs here, especially since I don't know how share URLs are handled by the driver or if that's something office-web-host is currently handling

Reply:
@christiango I totally missed that part, going to add it and also change the interface the way you suggested

Reply:
Also since now we merged the app url resolver(OdspDriverUrlResolver2) with the driver url resolver(OdspdriverUrlResolver), you need to make changes to "2" one for making it to nav param.,